src/main/java/net/minestom/server/instance/lighting/ChunkLight.java

```diff
@@ -0,0 +1,155 @@
+package net.minestom.server.instance.lighting;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.instance.Chunk;
+import net.minestom.server.network.packet.server.play.UpdateLightPacket;
+
+/**
+ * Configure chunk light data.
+ * <p>
+ * Notably used in {@link ChunkLightEngine#lightChunk(ChunkLight)} when implementing {@link ChunkLightEngine}
+ */
+public class ChunkLight {
+
+    // Data
+    final Chunk chunk;
+
+    // List of sky's light levels
+    List<byte[]> skyLight;
+
+    // List of block's light levels
+    List<byte[]> blockLight;
+
+    public ChunkLight(Chunk chunk) {
+        this.chunk = chunk;
+
+        // Create a fixed sized list with a length of 18.
+        this.skyLight = Arrays.asList(new byte[18][2048]);
+        this.blockLight = Arrays.asList(new byte[18][2048]);
+    }
+
+    /**
+     * Update the light level for a block at the given position
+     * <p>
+     * Ensure each coordinate is less than 16.
+     * 
+     * @param x Chunk block X
+     * @param y Chunk block Y
+     * @param z Chunk block Z
+     * @param blockLightLevel Light emission level from 0 to 15
+     */
+    public void setBlockLight(int x, int y, int z, int blockLightLevel) {
+        setLight(x, y, z, blockLightLevel, true);
+    }
+
+    /**
+     * Update the light level for the sky at the given position
+     * <p>
+     * Ensure each coordinate is less than 16.
+     * 
+     * @param x Chunk block X
+     * @param y Chunk block Y
+     * @param z Chunk block Z
+     * @param skyLightLevel Light emission level from 0 to 15
+     */
+    public void setSkyLight(int x, int y, int z, int skyLightLevel) {
+        setLight(x, y, z, skyLightLevel, false);
+    }
+
+    // Sky and block uses same exact parser
+    private void setLight(int x, int y, int z, int lightLevel, boolean isBlock) {
```

Comment:
I am unsure whats happening here. `y` is being used for an index in the chunk section as well as the chunk section itself. This seems incorrect, unless I am missing something.

Reply:
This may of been a blunder on my part.

The class was originally meant to handle per-chunk instead of per-chunk-section. With that stated, my intention was to read a raw coordinate (-1,256) and get the chunk section and chunk coordinate Y based on the info. So 33 would be index 2 at a Y value of 1.